fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -1008,6 +1009,45 @@ ACTOR Future<Void> dataDistributionTracker(Reference<InitialDataDistribution> in
 			when(Promise<int64_t> req = waitNext(getAverageShardBytes)) {
 				req.send(self.maxShardSize->get().get() / 2);
 			}
+			when(ServerTeamInfo req = waitNext(triggerSplitForStorageQueueTooLong)) {
+				TraceEvent e("DDTrackerStorageServerQueueTooLongNotified", self.distributorId);
+				e.detail("Server", req.serverId);
+				e.detail("Teams", req.teams.size());
+				int64_t maxShardWriteTraffic = 0;
+				KeyRange shardToMove;
+				ShardsAffectedByTeamFailure::Team selectedTeam;
+				for (const auto& team : req.teams) {
+					for (auto const& shard : self.shardsAffectedByTeamFailure->getShardsFor(team)) {
+						for (auto it : self.shards.intersectingRanges(shard)) {
+							if (it->value().stats->get().present()) {
+								int64_t shardWriteTraffic = it->value().stats->get().get().metrics.bytesPerKSecond;
+								if (shardWriteTraffic > maxShardWriteTraffic) {
+									shardToMove = it->range();
+									selectedTeam = team;
+									maxShardWriteTraffic = shardWriteTraffic;
+								}
+							}
+						}
+					}
+				}
+				if (!shardToMove.empty()) {
```

Comment:
Possibly more concise to change line 1024 to 
`if (shardWriteTraffic > maxShardWriteTraffic && shardWriteTraffic > SERVER_KNOBS->DD_MIN_SHARD_BYTES_PER_KSEC_TO_MOVE_OUT)`
and remove lines 1034 to 1037